tests/components/smarttub/test_init.py

```diff
@@ -3,46 +3,44 @@
 import asyncio
 from unittest.mock import patch
 
-import pytest
 from smarttub import LoginFailed
 
 from homeassistant.components import smarttub
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.config_entries import ENTRY_STATE_SETUP_RETRY
 from homeassistant.setup import async_setup_component
 
 
-async def test_setup_with_no_config(hass):
+async def test_setup_with_no_config(setup_component, hass, smarttub_api):
     """Test that we do not discover anything."""
-    assert await async_setup_component(hass, smarttub.DOMAIN, {}) is True
 
     # No flows started
     assert len(hass.config_entries.flow.async_progress()) == 0
 
-    assert smarttub.const.SMARTTUB_CONTROLLER not in hass.data[smarttub.DOMAIN]
+    smarttub_api.login.assert_not_called()
 
 
-async def test_setup_entry_not_ready(hass, config_entry, smarttub_api):
+async def test_setup_entry_not_ready(setup_component, hass, config_entry, smarttub_api):
     """Test setup when the entry is not ready."""
-    assert await async_setup_component(hass, smarttub.DOMAIN, {}) is True
     smarttub_api.login.side_effect = asyncio.TimeoutError
 
-    with pytest.raises(ConfigEntryNotReady):
-        await smarttub.async_setup_entry(hass, config_entry)
+    config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    assert config_entry.state == ENTRY_STATE_SETUP_RETRY
 
 
-async def test_setup_auth_failed(hass, config_entry, smarttub_api):
+async def test_setup_auth_failed(setup_component, hass, config_entry, smarttub_api):
     """Test setup when the credentials are invalid."""
-    assert await async_setup_component(hass, smarttub.DOMAIN, {}) is True
     smarttub_api.login.side_effect = LoginFailed
 
     assert await smarttub.async_setup_entry(hass, config_entry) is False
```

Comment:
Please replace all of these with `hass.config_entries.async_setup` and for the unload the corresponding config entries method.

Reply:
Sorry, missed that one.  Fixed